### ISSUE TITLE
Add annotations for doing static analysis using mypy.

### ISSUE THREAD

**Issue by ashwch** [zentropi/python-zentropi#4]
This will help in catching bugs earlier and will make the code more readable as we will know what a function expects and what it will return.

**Comment by hiway** [comment]
Yes, we already have mypy testing set up with tox ( `tox -e typecheck` ) - adding type hints and removing some ignored bits is required.

**Comment by ashwch** [comment]
@hiway I can take this over if you're already not working on it.

**Comment by hiway** [comment]
@ashwch that would be wonderful!
